pkg/scheduler/scheduler.go

```diff
@@ -381,12 +394,13 @@ func fitsFlavorLimits(name corev1.ResourceName, val int64, cq *cache.ClusterQueu
 	if borrow < 0 {
 		borrow = 0
 	}
-	if cohortUsed+val > cohortTotal {
-		// Doesn't fit even with borrowing.
+
+	lack := cohortUsed + val - cohortTotal
+	if lack > 0 {
 		// TODO(PostMVP): preemption could help if borrow == 0
-		return false, 0
+		return false, 0, fmt.Errorf("flavor %s still lack %d after borrowing", flavor.Name, lack)
```

Comment:
If there is no cohort, then there is no borrowing. Let's distinguish the 2 cases in the message.